### DISCUSSION TITLE
Increase Session Duration/Timeout ?

### DISCUSSION THREAD

**Discussion by abhchand** [mealie-recipes/mealie#729] [Ideas]
I noticed that Mealie usually logs you out automatically after a few hours (not sure what the session value is set to). 

Are people open to extending this duration? My suggestion is 1 month - most services let you stay logged in for an extended period of time.

The issue I'm facing is that my highly non-technical family finds it difficult to keep logging in every day, and it prevents over half of them from using it.

Great app! Appreciate all the developers contributing here.

**Comment by hay-kot** [comment]
You can set 'TOKEN_TIME' to whatever you'd like to extend it.

See https://hay-kot.github.io/mealie/documentation/getting-started/install/#env-variables

Although, I agree the default is too short.

**Reply by abhchand** [reply]
Ah i didn't realize that. Thanks!

**Comment by ChipSkylark37** [comment]
Bringing this back up - Would it be possible to bring the Token_Timeout into the Web UI vs Backend Config?

I am running this via Docker / Portainer and though its not a huge hassle, the only way to change these environmental variables is to completely trash and rebuild the Docker Container.

I am always hesitant to do this and run backups before, but it would be a lot simpler to have this option in the GUI for a selectable timeout (7 days, 2 weeks, etc).

**Reply by boc-the-git** [reply]
Hey @ChipSkylark37, whilst this could potentially be added to the UI, I'd suggest it's a reasonable amount of work for 0 material gain.

I'm curious what your concern is re destroying your container? If you set things up correctly you should be comfortable doing that 5 times a day! (the only issue that would give you is the ~1min of downtime in each rebuild)

If you don't have a reliable setup like that, then it's probably good to understand why, and get that sorted. Jump into the Discord and I'm sure people can help you out.

**Reply by ChipSkylark37** [reply]
Hey there. Thanks for the reply. 

You are correct that it shouldn’t be an issue to destroy and there’s minimal “downtime” as it’s recreating. 

It feels sort of reminiscent to using HomeAssistant in its early days - everything was YAML, and you had to code it into config files. Then, eventually everything was migrated into the GUI for ease of access and simplicity. 

I do agree its maybe not that important and I don’t see changing it often, but anything that takes it out of having to break out a computer to recreate the docker and instead can be a setting inside the container / GUI makes it that much more friendly to use. 

perhaps it’s just a feature request - I’ve already made the change in the ENV and recreated. Cheers!

**Comment by SamKook** [comment]
But for all of us who don't know how to modify docker containers, it's a lot more work to learn how to do it for infinitely more people, when a config should not be much work at all.
That would be the benefit, not have many people do work that could be done once and never again.
